src/components/hv-view/index.js

```diff
@@ -41,142 +47,213 @@ export default class HvView extends PureComponent<HvComponentProps> {
 
   props: HvComponentProps;
 
-  render() {
-    let viewOptions = this.props.options;
-    const { skipHref } = viewOptions || {};
-    const props: InternalProps = createProps(
+  attributes: Attributes;
+
+  constructor(props: HvComponentProps) {
+    super(props);
+    this.updateAttributes();
+  }
+
+  componentDidUpdate(prevProps: HvComponentProps) {
+    if (prevProps.element === this.props.element) {
+      return;
+    }
+
+    this.updateAttributes();
+  }
+
+  updateAttributes = () => {
+    // $FlowFixMe: reduce returns a mixed type, not Attributes
+    this.attributes = Object.values(ATTRIBUTES).reduce(
+      (attributes, name: string) => ({
+        ...attributes,
+        [name]: this.props.element.getAttribute(name),
+      }),
+      {},
+    );
+  };
+
+  hasInputFields = (): boolean => {
+    const textFields = this.props.element.getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      'text-field',
+    );
+    const textAreas = this.props.element.getElementsByTagNameNS(
+      Namespaces.HYPERVIEW,
+      'text-area',
+    );
+    return textFields.length > 0 || textAreas.length > 0;
+  };
+
+  getCommonProps = (): CommonProps => {
+    const style = createStyleProp(
       this.props.element,
       this.props.stylesheets,
-      viewOptions,
+      this.props.options,
     );
-    const scrollable = !!this.props.element.getAttribute('scroll');
+    const id = this.props.element.getAttribute('id');
+    if (!id) {
+      return { style };
+    }
+    if (Platform.OS === 'ios') {
+      return { style, testID: id };
+    }
+    return { accessibilityLabel: id, style };
```

Comment:
Why does the id need different name for iOS vs. Android?

Reply:
I don't remember the exact details about this, but here's some context
- https://github.com/Instawork/hyperview/pull/227
- https://github.com/Instawork/hyperview/pull/93